File: HackerRank/Data Structure/SparseArray.cs
//https://www.hackerrank.com/challenges/sparse-arrays/problem
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

namespace HackerRank
{
    public class SparseArray
    {
        public SparseArray()
        {
        }
        static int[] othersolution(string[] strings, string[] queries)
        {
            int idx = 0;
            int[] result = new int[queries.Length];
            foreach (string query in queries)
            {
                result[idx++] = strings.Where(temp => temp == query).Count();
            }
            return result;
        }

        static int[] matchingStrings(string[] strings, string[] queries)
        {
            int[] result = new int[queries.Length];
            int idx = 0;
            foreach (string query in queries)
            {
                int counter = 0;
                
                foreach (string str in strings)
                {
                    if (query == str)
                    {
                        counter += 1;
                    }
                }
                result[idx] = counter;
                ++idx;
            }
            return result;
        }
        static void Main(string[] args)
        {
           // TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

            int stringsCount = Convert.ToInt32(Console.ReadLine());

            string[] strings = new string[stringsCount];

            for (int i = 0; i < stringsCount; i++)
            {
                string stringsItem = Console.ReadLine();
                strings[i] = stringsItem;
            }

            int queriesCount = Convert.ToInt32(Console.ReadLine());

            string[] queries = new string[queriesCount];

            for (int i = 0; i < queriesCount; i++)
            {
                string queriesItem = Console.ReadLine();
                queries[i] = queriesItem;
            }

            int[] res = matchingStrings(strings, queries);

            Array.ForEach(res, Console.WriteLine);
           // textWriter.WriteLine(string.Join("\n", res));

           // textWriter.Flush();
           // textWriter.Close();
        }
    }
}

File: HackerRank/Data Structure/SparseArray.py
# https://www.hackerrank.com/challenges/sparse-arrays/problem?h_r=next-challenge&h_v=zen
import math
import os
import random
import re
import sys


def matchingStrings(strings, queries):
    result = []
    for query in queries:
        counter = 0
        for string in strings:

            if string == query:
                counter += 1

        result.append(counter)
    return result


def otherAnswer(strings, queries):
    return [strings.count(q) for q in queries]



if __name__ == '__main__':
    #fptr = open(os.environ['OUTPUT_PATH'], 'w')

    strings_count = int(input())

    strings = []

    for _ in range(strings_count):
        strings_item = input()
        strings.append(strings_item)

    queries_count = int(input())

    queries = []

    for _ in range(queries_count):
        queries_item = input()
        queries.append(queries_item)

    res = matchingStrings(strings, queries)
    res2 = otherAnswer(strings, queries)
    print(res)
    print(res2)
    #fptr.write('\n'.join(map(str, res)))
    #fptr.write('\n')

    #fptr.close()